src/Kiota.Builder/CodeDOM/CodeNamespace.cs

```diff
@@ -29,6 +30,18 @@ public override string Name
             StartBlock.Name = name;
         }
     }
+
+    public CodeFile FindFileOrInitializeWith(string fileName, params CodeElement[] children)
+    {
+        if (FindChildByName<CodeFile>(fileName, false) is CodeFile existingFile)
+            return existingFile;
+
+        var file = new CodeFile(fileName, children);
```

Comment:
should we also remove the children when present in the namespace to avoid having them present twice in the code dom?